components/github-api-client/src/client.rs

```diff
@@ -258,286 +259,94 @@ impl GitHubClient {
-    pub blog: Option<String>,
-    pub location: Option<String>,
-    pub email: Option<String>,
-    pub hireable: Option<bool>,
-    pub bio: Option<String>,
-    pub public_repos: Option<u32>,
-    pub public_gists: Option<u32>,
-    pub followers: Option<u32>,
-    pub following: Option<u32>,
-    pub created_at: Option<String>,
-    pub updated_at: Option<String>,
-}
-
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Email {
-    pub email: String,
-    pub primary: bool,
-    pub verified: bool,
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-pub struct AuthOk {
-    pub access_token: String,
-    pub scope: String,
-    pub token_type: String,
-}
-
-impl AuthOk {
-    pub fn missing_auth_scopes(&self) -> Vec<&'static str> {
-        let mut scopes = vec![];
-        for scope in AUTH_SCOPES.iter() {
-            if !self.scope.split(",").collect::<Vec<&str>>().iter().any(
-                |p| {
-                    p == scope
-                },
-            )
-            {
-                scopes.push(*scope);
-            }
-        }
-        scopes
-    }
 }
 
-#[derive(Deserialize, Serialize, Debug)]
-pub struct AuthErr {
-    pub error: String,
-    pub error_description: String,
-    pub error_uri: String,
-}
-
-impl fmt::Display for AuthErr {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(
-            f,
-            "err={}, desc={}, uri={}",
-            self.error,
-            self.error_description,
-            self.error_uri
-        )
-    }
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-pub struct Search {
-    total_count: u64,
-    incomplete_results: bool,
-    items: Vec<SearchItem>,
-}
-
-#[derive(Debug, Deserialize, Serialize)]
-pub struct SearchItem {
-    name: String,
-    path: PathBuf,
-    sha: String,
-    url: String,
-    git_url: String,
-    html_url: String,
-    repository: Repository,
-    score: f64,
-}
-
-#[derive(Deserialize, Serialize)]
-pub struct InstallationAccessToken {
-    pub token: String,
-    pub expires_at: String,
-}
-
-#[derive(Deserialize, Serialize, Debug)]
-pub struct InstallationAuthErr {
-    pub message: String,
-    pub documentation_url: String,
-}
-
-impl fmt::Display for InstallationAuthErr {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(
-            f,
-            "message={}, documentation_url={}",
-            self.message,
-            self.documentation_url
-        )
-    }
+fn generate_app_token<T>(key_path: T) -> String
+where
+    T: ToString,
+{
+    let mut payload = jwt::Payload::new();
+    let header = jwt::Header::new(jwt::Algorithm::RS256);
+    let now = SystemTime::now().duration_since(UNIX_EPOCH).unwrap();
+    let expiration = now + Duration::from_secs(10 * 60);
+    payload.insert("iat".to_string(), now.as_secs().to_string());
+    payload.insert("exp".to_string(), expiration.as_secs().to_string());
+    payload.insert("iss".to_string(), 5565.to_string());
```

Comment:
did we intend to hard code the app id?

Reply:
Absolutely not, I've got this fixed in another branch ;)